### PR TITLE
Add support for Rails TZ handling

### PR DESCRIPTION
**Pull request by ares**
@adamruzicka mind to take a look? Not sure how to properly test this since ActiveSupport::TimeZone is not a dependency.

### CODE REVIEW

File: lib/scoped_search/definition.rb [hunk 1]
```diff
@@ -278,6 +278,8 @@ def default_fields_for(value, operator = nil)
     # Try to parse a string as a datetime.
     # Supported formats are Today, Yesterday, Sunday, '1 day ago', '2 hours ago', '3 months ago', '4 weeks from now', 'Jan 23, 2004'
     # And many more formats that are documented in Ruby DateTime API Doc.
+    # In case Time responds to #zone, we know this is Rails environment and we can use Time.zone.parse. The benefir is that the
```
Comment (adamruzicka) [inline lib/scoped_search/definition.rb]: `s/benefir/benefit`

File: lib/scoped_search/definition.rb [hunk 2]
```diff
@@ -286,7 +288,11 @@ def parse_temporal(value)
       return (eval($1.strip.gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bago)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_datetime if value =~ /\A\s*(\d+\s+\b(?:hours?|minutes?)\b\s+\bfrom\s+now)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bfrom\s+now)\b\s*\z/i
-      DateTime.parse(value, true) rescue nil
+      if Time.respond_to?(:zone)
+        Time.zone.parse(value)
+      else
+        DateTime.parse(value, true) rescue nil
```
Comment (adamruzicka) [inline lib/scoped_search/definition.rb]: We should rescue the `Time.zone` branch too

File: lib/scoped_search/definition.rb [hunk 3]
```diff
@@ -286,7 +288,11 @@ def parse_temporal(value)
       return (eval($1.strip.gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bago)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_datetime if value =~ /\A\s*(\d+\s+\b(?:hours?|minutes?)\b\s+\bfrom\s+now)\b\s*\z/i
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bfrom\s+now)\b\s*\z/i
-      DateTime.parse(value, true) rescue nil
+      if Time.respond_to?(:zone)
+        Time.zone.parse(value)
```
Comment (adamruzicka) [inline lib/scoped_search/definition.rb]: Maybe add `.to_datetime` at the end to always return `DateTime`?

File: lib/scoped_search/definition.rb (hunk 2 above)
Comment (ares) [reply lib/scoped_search/definition.rb]: in case of wrong format it does not raise exception but returns nil

Comment (adamruzicka) [reply lib/scoped_search/definition.rb]: Not always
```ruby
pry(main)> Time.zone.parse('123:15:00')
ArgumentError: argument out of range
```

Comment (ares) [reply lib/scoped_search/definition.rb]: interesting, updated

**Comment by ares** [comment]
travis is no longer triggered for PRs, @wvanbergen do you happen to know why? Anyway, this should fix all comments and add testing with active support

**Comment by adamruzicka** [comment]
Looks good to me, waiting for Travis to confirm

**Comment by ares** [comment]
Since I don't have admin permissions over the repo, I can't look at the hooks configuration. @wvanbergen could you please take a look?

**Comment by wvanbergen** [comment]
Looks like it was deactivated on Travis's end. I think I have re-enabled it. Can you try by pushing something to this branch?

**Comment by ares** [comment]
Thanks @wvanbergen, it seems travis is now running again!

**Comment by adamruzicka** [comment]
Travis is running but doesn't like the changes

**Comment by ares** [comment]
fixed the travis definition, it seems now that the change has some negative effect, I'll need to look into more details, seems that the change affected even gemsets without active support

**Comment by ares** [comment]
Tests are green now, I updated travis configuration too and removed some old rubies to make the test matrix smaller. I think this is ready to go in. Btw tests revealed an issue if Time.zone is nil (no timezone set) so fixed that as well.

**Comment by ares** [comment]
@wvanbergen any concern with old rubies drop from travis matrix? I think we really don't need them anymore

**Comment by wvanbergen** [comment]
Can we still support ruby 2.3.7? That's the version macOS ships with.

**Comment by ares** [comment]
Fair enough, in fact RHEL 7 still ships Ruby 2.0 so I kept old version. The matrix is now bigger but travis seems to handle it fine. Let's see if all turn green now

**Comment by ares** [comment]
OK, all green now including old rubies :-)

**Comment by ares** [comment]
could this be merged pls?

**Comment by adamruzicka** [comment]
Thanks @ares !

**Comment by ares** [comment]
Thank you!

**Pull request merged**